executor/adapter.go

```diff
@@ -1171,6 +1171,17 @@ func getEncodedPlan(sctx sessionctx.Context, p plannercore.Plan, genHint bool) (
 	}
 	if genHint {
 		hints := plannercore.GenHintsFromPhysicalPlan(p)
+		for _, tableHint := range sctx.GetSessionVars().StmtCtx.OriginalTableHints {
+			// some hints like 'memory_quota' cannot be extracted from the PhysicalPlan directly,
+			// so we have to iterate all hints from the customer and keep some other necessary hints.
+			switch tableHint.HintName.L {
+			case "memory_quota", "use_toja", "no_index_merge", "max_execution_time",
+				plannercore.HintAggToCop, plannercore.HintIgnoreIndex,
```

Comment:
Besides, if some hints can not work should we store them? For example, if we use the `HintLimitToCop`, but there is no limit or the limit can not be pushed down.

Reply:
Yes, theoretically speaking, we shouldn't keep `HintLimitToCop` in this case, but it's not easy to implement it correctly and simply over the current codebase. So I tend to keep `HintLimitToCop` now for safety and leave this as a further optimization.